### DISCUSSION TITLE
Saving to deploy to production

### DISCUSSION THREAD

**Discussion by JosephTLucas** [manujosephv/pytorch_tabular#57] [Q&A] [answered]
`tabular_model.save_model("examples/basic")` saves a lot of large objects that would be necessary to resume training. What’s the minimal set that could be deployed to production for predictions? I don’t want to track/move checkpoints. Can I just use ONNX or load a state dict? I’m not sure what general PyTorch methods I can use and what functionality I’d lose.

**Comment by manujosephv** [comment, marked as answer]
PyTorch Tabular only saves the files which are necessary for reconstruction, even if it is for inference. Apart from the model (which is the `.ckpt`), it also needs the necessary preprocessing of the data(which is `datamodule.sav`), the config (`config.yml`), etc. to do inference.

The large object might be the `datamodule.sav`, which is because we are holding on to the data in the datamodule. That is something that we can remove to make the whole folder size lesser. Will add that to the roadmap to make this experience smoother.

**Reply by JosephTLucas** [reply]
Thanks @manujosephv. It is indeed a huge datamodule. I’m happy to help test, document, or anything else.

**Reply by manujosephv** [reply]
@JosephTLucas  This feature (exporting pure PyTorch model, or as ONNX(although experimental right now)) is currently part of PyTorch Tabular. In addition, also more flexibility in working with the components, saving different parts of the pipeline separately etc.
